nixos/modules/services/networking/sabnzbd/default.nix

```diff
@@ -0,0 +1,609 @@
+                      example = 50;
+                      default = 8;
+                    };
+                    timeout = mkOption {
+                      type = types.int;
+                      description = ''
+                        Time, in seconds, to wait for a response before
+                        attempting error recovery.
+                      '';
+                      default = 60;
+                    };
+                    ssl = mkOption {
+                      type = types.bool;
+                      description = ''
+                        Whether the server supports TLS
+                      '';
+                      default = true;
+                    };
+                    ssl_verify = mkOption {
+                      type =
+                        types.coercedTo
+                          (types.enum [
+                            "strict"
+                            "allow injection"
+                            "none"
+                          ])
+                          (
+                            val:
+                            if val == "strict" then
+                              3
+                            else if val == "allow injection" then
+                              2
+                            else
+                              0
+                          )
+                          (
+                            types.enum [
+                              0
+                              1
+                              2
+                              3
+                            ]
+                          );
+
+                      description = ''
+                        Level of TLS verification. Supported values:
+                        3, 'strict'          -- strict (normal) verification
+                        2, 'allow injection' -- allow locally injected certificates
+                        1, 0, 'none'         -- no verification
+                      '';
+                      default = "strict";
+                    };
+                    expire_date = mkOption {
+                      type = types.nullOr types.str;
+                      description = ''
+                        If Notifications are enabled and an expiry date is
+                        set, warn 5 days before expiry. This setting
+                        does not automatically disable the server.
+                        Expected format: yyyy-mm-dd
+                      '';
+                      default = null;
+                    };
+                  };
+                }
+              );
+            };
+          };
+        };
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    warnings =
+      if cfg.configFile != null then
+        [
+          ''
+            `sabnzbd.configFile` is deprecated, consider using `sabnzbd.settings` instead.
+            If you have values set in `sabnzbd.settings` set, they will be ignored.
```

Comment:
This reads like the setting is ineffective which means it should be an assertion. But I wonder why we don't just use mkRemovedOptionModule instead then?

Reply:
I tried to avoid having an assertion that would break existing configs. I originally tried to have a deprecation warning if `configFile` was used, and an assertion that at least one of the two options was unused, but I couldn't get the latter to work, since with the default options in `settings` it appears to be always populated.